src/webots/nodes/WbWorldInfo.cpp

```diff
@@ -292,26 +299,32 @@ void WbWorldInfo::applyToOdeGlobalDamping() {
 
 // Computes an orthonormal basis whose 'yaw unit vector' is the opposite of the normalized gravity vector
 void WbWorldInfo::updateGravityBasis() {
-  if (!gravity().isNull()) {
-    mGravityUnitVector = gravity().normalized();
-    WbMathsUtilities::orthoBasis(-gravity(), mGravityBasis);
-  } else {
-    mGravityUnitVector.setXyz(0.0, -1.0, 0.0);
-    mGravityBasis[X].setXyz(1.0, 0.0, 0.0);
-    mGravityBasis[Y].setXyz(0.0, 1.0, 0.0);
-    mGravityBasis[Z].setXyz(0.0, 0.0, 1.0);
+  if (mCoordinateSystem->value() == "ENU") {
+    mEastVector = WbVector3(1, 0, 0);
+    mNorthVector = WbVector3(0, 1, 0);
+    mUpVector = WbVector3(0, 0, 1);
+    mGravityBasis[X].setXyz(0, 1, 0);
+    mGravityBasis[Y].setXyz(0, 0, 1);
+    mGravityBasis[Z].setXyz(1, 0, 0);
+  } else {  // "NUE"
+    mNorthVector = WbVector3(1, 0, 0);
+    mUpVector = WbVector3(0, 1, 0);
+    mEastVector = WbVector3(0, 0, 1);
+    mGravityBasis[X].setXyz(1, 0, 0);
+    mGravityBasis[Y].setXyz(0, 1, 0);
+    mGravityBasis[Z].setXyz(0, 0, 1);
   }
+  mGravityUnitVector = -mUpVector;
+  mGravityVector = mGravityUnitVector * mGravity->value();
 }
 
-void WbWorldInfo::updateNorthDirection() {
-  if (mNorthDirection->value().isNull()) {
-    mNorthDirection->setValue(1, 0, 0);
-    parsingWarn(tr("'northDirection' must be a unit vector. Reset to default value (1, 0, 0)."));
-  }
+void WbWorldInfo::updateCoordinateSystem() {
+  warn(tr("Please save and revert the world so that the change of coordinate system is taken into account when reloading "
+          "procedural PROTOs."));
```

Comment:
```suggestion
          "procedural PROTO nodes."));
```

We decided a while ago that we should always use PROTO with a name (e.g. PROTO node, PROTO file, etc.).